### DISCUSSION TITLE
Typescript - onScroll argument types don't match

### DISCUSSION THREAD

**Discussion by migueSNM** [Comcast/react-data-grid#3438] [Q&A]
Hi there!

I've encountered an issue related to typescript when providing a callback function to the onScroll prop. The typescript error suggests that the prop expects a callback with an argument of type ScrollState, but the current handler of DataGrid component returns a React.UIEvent<HTMLElement> (HTML event).

I'm using the following versions:

`"@types/react-data-grid": "^4.0.12",`
`"react-data-grid": "^7.0.0-beta.42",`

Has anyone else faced this issue, and if so, what workarounds or best practices have you employed to resolve it without resorting to using the any type?

Thanks for the help!

**Comment by nstepien** [comment]
`react-data-grid` ships with types included, so you should uninstall `@types/react-data-grid`
